src/stages/glue-transform.ts

```diff
@@ -0,0 +1,68 @@
+import * as glueAlpha from '@aws-cdk/aws-glue-alpha';
+import * as glue from 'aws-cdk-lib/aws-glue';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+import { StateMachineStage, StateMachineStageProps } from '../pipelines/stage';
+
+export interface GlueTransformStageProps extends StateMachineStageProps {
+  readonly job?: glueAlpha.Job;
+  readonly jobName: string;
+  readonly jobArgs?: { [key: string]: any };
+  readonly crawlerName?: string;
+  readonly cfnCrawler?: glue.CfnCrawler;
+}
+
+export class GlueTransformStage extends StateMachineStage {
+  private readonly job?: glueAlpha.Job;
+  private readonly jobName?: string;
+  private readonly jobArgs?: { [key: string]: any };
+  private readonly crawlerName?: string;
+  private readonly cfnCrawler?: glue.CfnCrawler;
+
+  constructor(scope: Construct, id: string, props: GlueTransformStageProps) {
+    super(scope, id, props);
+
+    this.job = props.job;
+    this.jobName = props.jobName;
+    this.jobArgs = props.jobArgs;
+    this.crawlerName = props.crawlerName;
+    this.cfnCrawler = props.cfnCrawler;
+
+    if (this.jobName == undefined && this.job == undefined) {
+      throw TypeError("Either 'jobName' or 'job' is required");
+    }
+    if (this.crawlerName == undefined && this.cfnCrawler == undefined) {
+      throw TypeError("Either 'crawlerName' or 'cfnCrawler' is required");
+    }
+  }
+
+  protected createStateMachineSteps(): sfn.IChainable {
+
+    const startJobRun = new tasks.GlueStartJobRun(
+      this,
+      'Start Job Run',
+      {
+        glueJobName: this.jobName ?? this.job?.jobName ?? '',
```

Comment:
Why do we have the last null coalescing? If this string is empty, wouldn't the step just be broken? I think that would would be better is that we just store `this.jobName` as a class property, and make it just a `string` rather than `string?`. We can assign the value in the constructor, when we're also checking for null values. Then here, we can just reference `this.jobName`.

Reply:
Yeah since the typing isn't smart enough to understand I would throw an exception if either `jobName` or `job` itself is empty, this was just a workaround. Anyways since we'll always have `jobName` now this isn't needed